app/src/main/java/mozilla/lockbox/presenter/ItemDetailPresenter.kt

```diff
@@ -32,5 +76,13 @@ class ItemDetailPresenter(
                 }
                 .subscribe(view::updateItem)
                 .addTo(compositeDisposable)
+
+        isPasswordVisible = false
+    }
+
+    private fun clipboardCopy(label: String, str: String) {
+
+        val clip = ClipData.newPlainText(label, str)
+        this.clipboardManager.primaryClip = clip
```

Comment:
Thinking about this, I think I would prefer this to be in a `ClipboardStore` and we should communicate with this via the dispatcher. 

This will make telemetry and timed invalidation easier, and is conceptually similar to how the rest of the app is built.

Reply:
Please help on what is the correct way to include above code in a ClipboardStore, my problem is that clipboard manager is system Service that is obtained from Context, if I get the context in Fragment how is the correct way to pass the system service to the ClipboardStore

Reply:
Hmm. So, we need a `Context` in order to get a `ClipboardManager`, and the design is such that the `ClipboardStore` is being constructed away from the context, and thus the `android.text.ClipboardManager` system service. 

So, we have multiple choices here, none of them brilliant. 

 * an extension method on the companion object for `Application` to maintain singleton access `Application.shared`. We should be careful doing this, so we should watch out for trixyness in lifecycles and memory leaks.
 * Make the `ClipboardStore.clipboardManager` be optional, and inject it in from the `Application.onCreate` method.
 * Derive the `ClipboardManager` from the view in the presenter and pass it as part of the `Action` via the dispatcher to the `ClipboardStore` (noting that the `ClipboardStore` doesn't have to be directly referenced by the `ItemDetailPresenter`).